src/cdk/overlay/overlay-state.ts

```diff
@@ -53,6 +53,12 @@ export class OverlayState {
   /** The direction of the text in the overlay panel. */
   direction?: Direction = 'ltr';
 
+  constructor(state?: OverlayState) {
+    if (state) {
```

Comment:
I didn't use `extendObject`, because it would introduce a dependency between Material and the CDK.